coreLib/src/main/java/dev/kosmx/playerAnim/core/data/AnimationBinary.java

```diff
@@ -184,9 +184,11 @@ private static boolean readKeyframes(ByteBuffer buf, KeyframeAnimation.StateColl
             enabled = length >= 0;
         }
         for (int i = 0; i < length; i++) {
-
             int currentPos = buf.position();
-            if(! part.addKeyFrame(buf.getInt(), buf.getFloat(), Ease.getEase(buf.get()))){
+            int id = buf.getInt();
+            float value = buf.getFloat();
+            Ease ease = Ease.getEase(buf.get());
```

Comment:
What have you changed here? This is used to transfer the animation over the network and also pack it into an `.emotecraft` file, however you need to create a new version of the animation so you don't break anything, as far as I remember the latest one at the moment is 3.

Reply:
I wanted to have this work with Emotecraft so I started coding some stuff in this class but then I decided not to touch it
These were some remenants from my code which I have removed

Reply:
Well, for the future: you need to increase the number of `getCurrentVersion` and also check the passed `version` parameter of every method you modify in this class

Reply:
You need to do roughly the same thing in the `AnimationJson` class.

Reply:
> Well, for the future: you need to increase the `getCurrentVersion` and also check the passed `version` parameter of every method you modify in this class

I have done that actually but like I said I haven't tested it so I am not gonna try to get it merged yet
Thanks for all the help 🙏

Reply:
You can see how I did it in #98 (and yes, I look forward to your recommendations)